src/client/common/client_formatter.cpp

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) 2021 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "client_formatter.h"
+
+#include <multipass/cli/format_utils.h>
+#include <multipass/utils.h>
+
+#include <fmt/format.h>
+#include <yaml-cpp/yaml.h>
+
+#include <sstream>
+
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QJsonObject>
+
+namespace mp = multipass;
+namespace mpu = multipass::utils;
+
+std::string mp::ClientFormatter::format(const mp::AliasDict& aliases) const
+{
+    std::string formatted_output;
+
+    if (preferred_format == "csv")
+        formatted_output = format_csv(aliases);
+    else if (preferred_format == "json")
+        formatted_output = format_json(aliases);
+    else if (preferred_format == "table")
+        formatted_output = format_table(aliases);
+    else if (preferred_format == "yaml")
+        formatted_output = format_yaml(aliases);
+
+    return formatted_output;
+}
```

Comment:
I'm not a fan of all of the `if`/`else if`'s here.  In `format_utils.cpp`, it creates a map of the various formatters and then lookup is easy based on the preferred formatter.  I wonder if something like that would be better here.

Reply:
This is outdated, since the formatter was refactored.